beacon-chain/powchain/engine-api-client/v1/client.go

```diff
@@ -96,6 +96,10 @@ func New(ctx context.Context, endpoint string, opts ...Option) (*Client, error)
 
 // NewPayload calls the engine_newPayloadV1 method via JSON-RPC.
 func (c *Client) NewPayload(ctx context.Context, payload *pb.ExecutionPayload) ([]byte, error) {
+	start := time.Now()
+	defer func() {
+		newPayloadLatency.Observe(float64(time.Since(start).Milliseconds()))
+	}()
```

Comment:
Dumb question. What's the difference between?
```go
defer func() {
		newPayloadLatency.Observe(float64(time.Since(start).Milliseconds()))
	}()
```
and
```go
defer newPayloadLatency.Observe(float64(time.Since(start).Milliseconds()))
```

Reply:
your version is cleaner indeed

Reply:
They have different behavior. In @terencechain 's version, `time.Since(start).Milliseconds()` will execute immediately and only the overall call to `Observe` will be defered. In @rauljordan 's version, both will be deferred. Example of what I mean:
https://go.dev/play/p/-D6HLmmf0KP

Reply:
So the question would be, do you want to `.Observe` the time when the defer statement was reached (Terence's version), or at the time the function finished (Raul's version)?

Reply:
That Makes sense @kasey , thanks for the explanation!